products/views.py

```diff
@@ -27,6 +28,53 @@ def get(self, request):
 
         return JsonResponse({'MESSAGE': 'SUCCESS', 'RESULT': categories}, status=200) 
 
+class ProductView(View):
+      def get(self, request, **kwargs):
+
+        product = Product.objects.get(id=kwargs['product_id'])
+        category_id = product.category.id
+        price = product.price_set.get()
+        review = [
+            *product.category.product_set.filter(review__isnull=False).annotate(
+                content=F('review__content'),
+                star=F('review__star'),
+                date=F('review__created_at'),
+                user_name=F('review__user__name')
+            ).values(
+                'id', 'content', 'star', 'date', 'user_name'
+            )
+        ]
```

Comment:
연관 상품 리뷰 전부 불러오는 것 같은데 로직이 복잡해 보이네요.
아래 코드로 수정될 수 없을까요?
```suggestion
        reviews = list(Review.objects.filter(product__category_id=product.category.id).values('id', 'content', 'star'))
```

Reply:
가독성이 더 좋은거 같아요, 수정했습니다!